pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -461,6 +462,14 @@ func (e *EndpointController) checkLeftoverEndpoints() {
 	}
 	for i := range list.Items {
 		ep := &list.Items[i]
+		if _, ok := ep.Annotations[resourcelock.LeaderElectionRecordAnnotationKey]; ok {
```

Comment:
this seems inappropriate... I would have expected an empty service definition rather than changing the endpoints controller